CONTRIBUTING.adoc

```diff
@@ -108,55 +108,78 @@ NOTE: you must run your editor from the nix shell, and not from, say, an
-actually use flakes for normal Nix builds. To manage the dependencies,
-see https://nixos.org/manual/nix/unstable/command-ref/new-cli/nix3-flake.html#flake-inputs[the Nix flake inputs documentation]
-and https://nixos.org/manual/nix/unstable/command-ref/new-cli/nix3-flake-lock.html[the Nix flake lock command].
+You can use a ``source-repository-package`` stanza to pull in the
+unreleased version.
 
-Specifically, you will probabl want to say `nix flake lock --update-input <input-name>`.
-Do *not* use `nix flake update`, as that will update all the inputs, which we typically don't want to do.
+Try only to do this for a short time, as it does not play very well with
+tooling, and will interfere with the ability to release the node itself.
 
-Inside of the project nix-shell, a version of Nix with flakes is available under the alias `nix-flakes`.
+For packages that we do not control, we can end up in a situation where we
+have a fork that looks like it will be long-lived or permanent (e.g. the
+maintainer is unresponsive, or the change has been merged but not
+released).
 
-[[update-index-state]]
-=== How to update the Hackage index state
+In that case, release a patched version to the `Cardano package repository
+<https://github.com/input-output-hk/cardano-haskell-packages>`, which
+allows us to remove the ``source-repository-package`` stanza.
+
+See the README for instructions.
+
+=== Releasing a version of plutus-apps
+
+(There is much more to say here, this is just a small fragment)
+
+==== ... to the Cardano package repository
 
-The Hackage index state is pinned to a particular time in `cabal.project`.
-This helps with reproducibility: alongside using the same version of `cabal`, this ensures that everyone will get the same result from the `cabal` version solver.
-If you want to use a Hackage package from after the pinned index state time, you need to bump it.
-This is not a big deal, since all it does is change what packages `cabal` considers to be available when doing solving, but it *may* result in different versions being picked, so it's not completely safe.
+When releasing a new version of plutus-apps, all packages in this
+repository should be released to the `Cardano package repository
+<https://github.com/input-output-hk/cardano-haskell-packages>`.
 
-Note that `cabal` itself keeps track of what index states it knows about, so you may need to update this with `cabal update` in order for `cabal` to be happy.
+See the README for instructions, including a script to automate most of the
+process.
 
-The Nix code which builds our packages also cares about the index state.
-The set of index states which it knows about is controlled by `hackage.nix`, which is a Nix representation of Hackage.
-This therefore needs to be newer than the index state.
-You can update it xref:update-nix-pins[with the Nix flake commands].
+Please note that libraries need bounds on the version of their dependencies
+to avoid bitrot and be effectively reusable.
```

Comment:
We have a doc about release process: `doc/BRANCHING-AND-RELEASE.adoc`.

It should be updated with this.

Reply:
Thanks, done. I removed that part from CONTRIBUTING and added a sentence to doc/BRANCHING-AND-RELEASE.adoc